sw/airborne/modules/mission/mission_common.c

```diff
@@ -39,6 +39,14 @@ void mission_init(void)
   mission.insert_idx = 0;
   mission.current_idx = 0;
   mission.element_time = 0.;
+
+  // FIXME
```

Comment:
Can't you just check at the start of the `mission_register` if the initialisation is done, and else perform it?

Reply:
As long as the status of the memory at startup is not predictable, I don't see how you can tell if a non-zero value is legitimate or not.
I'm still working on the build system to integrate a dependency mechanism in the end. Then it will be possible to guarantee that mission is initialized for other modules using it. In the meantime, I don't have a clean solution.

Reply:
I though about adding a static value `initialised` from start which you set to `false`. Then you can check this during the adding.

Reply:
And to ensure that it is false fro . the start you can do something like this:
```
struct _mission mission = {
  .insert_idx = 0,
  .current_idx = 0,
  .element_time = 0,
  .initialised = 0
};
```

Reply:
That's an idea, thank. Actually it seems I can set the whole structure to 0 statically with `{0}`